File: src/Moryx.Operators/Attendances/IOperatorAssignable.cs
// Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using Moryx.AbstractionLayer.Capabilities;
using Moryx.AbstractionLayer.Resources;
using Moryx.Operators.Skills;

namespace Moryx.Operators.Attendances;

public interface IOperatorAssignable : IResource
{
    /// <summary>
    /// Skill required for an operator to work on this machine
    /// </summary>
    ICapabilities RequiredSkills { get; }

    /// <summary>
    /// Inform the assignable about the currently assigned operators and their skills
    /// </summary>
    /// <param name="attandances"></param>
    void AttendanceChanged(IReadOnlyList<AttendanceData> attandances);
}

public record AttendanceData(Operator Operator, IReadOnlyList<Skill> Skills);

File: src/Moryx.Shifts.Web/app/src/app/api/models/i-operator-assignable.ts
/*
 * Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/

/* tslint:disable */
/* eslint-disable */
import { ICapabilities } from '../models/i-capabilities';
export interface IOperatorAssignable {
  capabilities?: ICapabilities;
  id?: number;
  name?: string | null;
  requiredSkills?: ICapabilities;
}

